pkg/platform/secret.go

```diff
@@ -114,7 +117,23 @@ func ApplyPlatformSecret(
 				},
 			},
 			Data: secretPayload,
-		}, metav1.CreateOptions{})
+		}
+		if svcErr == nil {
+			// set virtual cluster service as an owner for platform secret, so it can be cleaned up by kubernetes GC controller
+			// once virtual cluster is uninstalled
+			secret.OwnerReferences = []metav1.OwnerReference{
```

Comment:
nit: we could also use SetOwnerReference from the [controllerutil library ](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.20.4/pkg/controller/controllerutil/controllerutil.go#L110)

Reply:
I initially used it, but then decided I want to set `BlockOwnerDeletion` to `false`, and in `SetOwnerReference` it's always set to true